packages/vm/lib/runBlock.ts

```diff
@@ -321,3 +337,26 @@ async function rewardAccount(state: StateManager, address: Buffer, reward: BN):
   account.balance = toBuffer(new BN(account.balance).add(reward))
   await state.putAccount(address, account)
 }
+
+// apply the DAO fork changes to the VM
+async function _applyDAOHardfork(state: StateManager) {
+  const DAORefundContractAddress = Buffer.from(DAORefundContract, 'hex')
+  if (!state.accountExists(DAORefundContractAddress)) {
+    await state.putAccount(DAORefundContractAddress, new Account())
+  }
+  const DAORefundAccount = await state.getAccount(DAORefundContractAddress)
+  let DAOBalance = new BN(DAORefundAccount.balance)
+
+  for (let address of DAOAccountList) {
+    // retrieve the account and add it to the DAO's Refund accounts' balance.
+    let account = await state.getAccount(Buffer.from(address, 'hex'))
+    DAOBalance.iadd(new BN(account.balance))
+    // clear the accounts' balance
+    account.balance = Buffer.alloc(0)
```

Comment:
I will likely never learn these kind of things 🤨 - they just seem to not fit in my head - but is it clear here that setting the account balance to zero is done with the empty Buffer (`[Buffer ]`) and not the zero Buffer (`[Buffer 00])`)?

Will not make it a blocker though since transition test runs from `ethereum/tests` seem to indicate that this is correct and leads to the correct transition state roots.

Reply:
This is worth checking out at some point if we would handle it correctly if we put a 1-byte zero-byte (😅 ). However to calculate the state root the actual Buffer should be empty and not Buffer.from('00','hex') (1 byte length) otherwise state root calculation gets messed up.